src/main/groovy/com/bmuschko/gradle/docker/tasks/DockerVersion.groovy

```diff
@@ -15,12 +15,12 @@
  */
 package com.bmuschko.gradle.docker.tasks
 
-import com.github.dockerjava.api.DockerClient
 import com.github.dockerjava.api.model.Version
 
 class DockerVersion extends AbstractDockerRemoteApiTask {
+
     @Override
-    void runRemoteCommand(DockerClient dockerClient) {
+    void runRemoteCommand() {
         logger.quiet "Retrieving Docker version."
         Version version = dockerClient.versionCmd().exec()
```

Comment:
Should this, and all other calls below, use `getDockerClient()` if only to keep the way we access things standard across the board?

Reply:
This is just Groovy syntactic sugar. It calls the getter method under the hood. If you rather want the Java style then we can make that happen though there's no functional benefit. I am not sure if we are even consistent across the board. It's unlikely.

Reply:
The only reason I note that is because the variable is also named `dockerClient`. I've no idea how groovy will make things work guess I'm just thinking it _may_ want to go directly to the variable instead of hitting the method? If that's not the case then disregard this comment.

Reply:
It calls the getter. Also the `dockerClient` variable is `private` so the subclasses can't access it.

Reply:
> private so the subclasses can't access 

In java maybe but with groovy ALL things are possible ;)

Reply:
I was wondering why remove the @Memoize annotation when creating dockerClient. I thought it was nice that the dockerClient gets reused for all the different needs as long as it is the same which is usually the case within the same program...

Reply:
Yeah, I guess we already used it that way for quite a while. I was concerned about changing values between build invocations. I am not sure how Gradle caches a plugin implementation which could lead to incorrect results. I restored it.